src/util/logger.cpp

```diff
@@ -0,0 +1,36 @@
+#include <cstring>
+
+#include "util/logger.h"
+
+
+namespace {
+
+const char* const kLogPreambleSuffix = " -";
+const std::size_t kLogPreambleSuffixLen = std::strlen(kLogPreambleSuffix);
+
+inline QByteArray preambleChars(const char* logContext) {
+    QByteArray preamble;
+    if (logContext != nullptr) {
+        std::size_t logContextLen = std::strlen(logContext);
```

Comment:
Did you consider to use here QLatin1String as well, to get around searching for \0?